enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -351,79 +351,67 @@ impl<'a> Handler<'a> {
     pub fn readv(&mut self) -> u64 {
         self.trace("readv", 3);
 
-        let fd = self.aex.gpr.rdi.raw();
-        let trusted = unsafe {
-            from_raw_parts_mut(
-                self.aex.gpr.rsi.raw() as *mut libc::iovec,
-                self.aex.gpr.rdx.raw() as usize,
-            )
-        };
+        let tiovec: &mut [libc::iovec] =
+            unsafe { self.aex.gpr.rsi.as_slice_mut(self.aex.gpr.rdx.into()) };
```

Comment:
In my version of this function, I avoid the wrap by inferring the type and passing the `tiovec` to `cursor.copy_slice()`.

Reply:
Please see below.